qemu/cpus.c

```diff
@@ -191,15 +169,10 @@ static int qemu_tcg_init_vcpu(CPUState *cpu)
         uc->tcg_halt_cond = cpu->halt_cond;
         snprintf(thread_name, VCPU_THREAD_NAME_SIZE, "CPU %d/TCG",
                 cpu->cpu_index);
-        if (qemu_thread_create(uc, cpu->thread, thread_name, qemu_tcg_cpu_thread_fn,
-                cpu, QEMU_THREAD_JOINABLE))
-            return -1;
+        qemu_thread_get_self(uc, cpu->thread);
```

Comment:
@aquynh that does not work as expected, due to the way the original code was laid out. At this point initialization is not done. My PR as submitted should even work with more than one CPU on a single Unicorn handle. If you put the CPU loop where the thread was, this flow happens:

1. init cpu
2. cpu loop
3. finish initing cpu

My submitted flow is:

1. init cpu
2. finish initing cpu
3. run cpu loop

In your suggested edit, if you have more than one CPU object on a uc handle for some reason, this happens:

1. init cpu
2. run cpu loop for all cpus
3. init cpu 2
4. run cpu loop for all cpus
5. finish initing cpus 1, 2

My flow is:

1. init cpu 1, 2
2. finish initing cpu 1, 2
3. run cpu loop for all cpus

Does that make sense?